### PR TITLE
Adding equipment from the manufacturer DATACOM

### PR DESCRIPTION
**Pull request by rogeriomendesp**

### CODE REVIEW

**Comment by rogeriomendesp** [comment]
image


Hi,

I've tried everything, I can't succeed.

any suggestion?

Thanks.

**Comment by rogeriomendesp** [comment]
image

**Comment by rogeriomendesp** [comment]
Hi,

I've tried everything, I can't succeed.

any suggestion?

Thanks.

**Review by rogeriomendesp — COMMENTED** [review]
Update

**Comment by ivareri** [comment]
@rogeriomendesp Seems like yamlfmt removes the `'` thats around the part_number, and splits the `comments:` into two lines. 

Try running `pre-commit run --files device-types/Datacom/DM4050-24GT-6XS.yaml`, followed by `git diff`.  Should show you the changes, if it looks OK you could just try to `git add` the file again and re-run the pre-commit check

**Comment by rogeriomendesp** [comment]
> pre-commit run --files device-types/Datacom/DM4050-24GT-6XS.yaml

image

**Comment by ivareri** [comment]
Did you follow the chapter on [Data Validation](https://github.com/netbox-community/devicetype-library#data-validation--commit-quality-checks) in the readme? 

Then something like this:

Screenshot

File: device-types/Datacom/DM4050-24GT-6XS.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+manufacturer: Datacom
+model: DM4050 24GT+6XS
+slug: dm4050-24gt-6xs
+part_number: 8005189xx
+u_height: 1.0
+is_full_depth: true
+airflow: side-to-rear
+comments: L2/L3 wirespeed switch with 24 electrical Gigabit Ethernet ports (RJ45SFP) and 6 optical 10Gisgabit Ethernet ports (SFP+)
+  1U Metal cabinet for 19-inch racks. Two slots for power supplies Power supplies and SFPs to be ordered separately (https://www.datacom.com.br/uploads/product/file/97aede61eba553f23060d983b6e68152.pdf)
```
Comment (ivareri) [inline device-types/Datacom/DM4050-24GT-6XS.yaml:10]: Seems like this is the change yamlfmt wants

```suggestion
comments: L2/L3 wirespeed switch with 24 electrical Gigabit Ethernet ports (RJ45SFP) and 6 optical 10Gisgabit Ethernet ports (SFP+) 1U Metal cabinet for
  19-inch racks. Two slots for power supplies Power supplies and SFPs to be ordered separately (https://www.datacom.com.br/uploads/product/file/97aede61eba553f23060d983b6e68152.pdf)

**Comment by rogeriomendesp** [comment]
> Você seguiu o capítulo sobre [validação de dados](https://github.com/netbox-community/devicetype-library#data-validation--commit-quality-checks) no leia-me?
> 
> Então algo assim:
> 
> Captura de tela

sorry.

I do it through github desktop.

**Comment by rogeriomendesp** [comment]
Hi,
Is there a way to pre-commit on github desktop?

Thanks.

**Comment by jsenecal** [comment]
> Hi, Is there a way to pre-commit on github desktop?

You also need a python environment to run the scripts, etc.
Know anything about vscode and wsl2 ?

Perhaps someone else can pick up where you left off and properly lint your files :)

**Comment by rogeriomendesp** [comment]
Hi

I'm changing the files, I'll pre-commit and then send them.

File: device-types/Datacom/DM4050-24GT-6XS.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
+---
+manufacturer: Datacom
+model: DM4050 24GT+6XS
+slug: dm4050-24gt-6xs
```
Comment (danner26) [inline device-types/Datacom/DM4050-24GT-6XS.yaml:4]: Please append `datacom-` to the slug name

File: device-types/Datacom/DM4050-24GT-6XS.yaml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+airflow: side-to-rear
+comments: L2/L3 wirespeed switch with 24 electrical Gigabit Ethernet ports (RJ45SFP) and 6 optical 10Gisgabit Ethernet ports (SFP+) 1U Metal cabinet for
+  19-inch racks. Two slots for power supplies Power supplies and SFPs to be ordered separately (https://www.datacom.com.br/uploads/product/file/97aede61eba553f23060d983b6e68152.pdf)
+weight: 2.96
+weight_unit: kg
+console-ports:
+  - name: Console
+    type: rj-45
+  - name: Console2
+    type: usb-mini-b
```
Comment (danner26) [inline device-types/Datacom/DM4050-24GT-6XS.yaml:17]: Is this console port actually used for console access? If not, please remove it.

**Review by danner26 — CHANGES_REQUESTED** [review]

**Comment by danner26** [comment]
Hello, we have released a new set of tests and features to assist with properly creating definition files for the DeviceType-Library. These tests will ensure that all files conform to a standardized format and contain valid data. Over the coming weeks we will be adding more tests to handle other currently undefined cases. If you notice anything that you believe is a bug, please @ me in a comment and details what you think is incorrect.

Along with additional test cases we have added a feature that will annotate why tests fail. To access these annotations after a test has completed, please navigate to the _Files changed_ tab within your PR. At the very bottom of the page you will see a list of errors along with the filename in question and the error which was found.

With all this being said, if your PR is validated then great! Please @ me me so I can review it further! If your PR is not valid, please review the issues and correct them; in the meantime I will mark your PR as a draft. After correcting the issues and recieving a valid status, please mark the PR as ready to be reviewed. Thank you!

**Comment by danner26** [comment]
Hey, thanks for reaching out. The first issue is that you are working directly on the master branch. I would suggest that you copy your changed files elsewhere, say to /tmp and then pull down the latest changes into your GitHub fork. Once that is done, do a `git fetch` and reset your master branch to the head commit (which should be the latest master commit on the upstream repo). After restoring the master branch, create a new branch off of the master. `git checkout -b <branch_name>`. Replace `<branch_name>` with the name youd like to call your branch. Then add your files back, commit and run the tests.

That should help resolve a lot of your issues  you are facing

**Pull request closed without merging**